lib/rules/prefer-explicit-assert.js

```diff
@@ -24,6 +24,12 @@ const isDeclared = node => node.parent.type === 'VariableDeclarator';
 const isReturnedByReturnStatement = node =>
   node.parent.type === 'ReturnStatement';
 
+const isInDestructuringStatement = node =>
+  (node.parent.type === 'Property' &&
+    node.parent.parent.type === 'ObjectPattern') ||
+  (node.parent.type === 'ArrayPattern' &&
+    node.parent.parent.type === 'VariableDeclarator');
```

Comment:
I don't think checking `VariableDeclarator` is necessary, is it?

Reply:
oh, I see, it's not. Fixed!

This made me think of something else, though. Should the following line be considered as a valid one? Right now it is consideren invalid:

`const a = [ getByText('foo'), 'bar' ]`

Reply:
That's interesting... I guess it should be considered as valid since we don't use it as an assertion here. We need to fix that as well.

Thanks for spotting the issue! 🙂

Reply:
okay, I'll work on that and update the PR accordingly! :)

Reply:
All right! Thanks for working on that 😄

Reply:
Okay, just updated the PR. I think checking if grandparent is an `ArrayExpression` will suffice, but please do doublecheck it as I'm relying on tests I wrote and the AST explorer which is just learned about when you shared the screenshots 😇 

thanks for helping me out!

Reply:
I got it working by doing this: 

1. Add a `findParent` function in the `utils` file:

```javascript

const findParent = (node, cb) => {
  if (cb(node)) {
    return node;
  } else if (node.parent) {
    return findParent(node.parent, cb);
  } else {
    return null;
  }
};
```

2. Modify both `isDeclared` and `findCallExpressionParent`:

```javascript
const findCallExpressionParent = node =>
  findParent(node, node => node.type === 'CallExpression');

// ... 

const isDeclared = node => {
  return !!findParent(node, node => node.type === 'VariableDeclarator');
};
```

3. I removed the `isInArrayDeclaration` function

4. I added a last test:

```javascript
    {
      code: `const a = { foo: getByText('bar') }`,
    },
```

I think it should be good by doing so 🙂

Reply:
It looks way smarter to check for positive cases rather than trying to take into account every posssible wrong edge case. Updating the PR.